src/hardware/sblaster.cpp

```diff
@@ -3117,308 +3154,282 @@ static bool is_cms_enabled(const SbType sbtype)
 
 void shutdown_sblaster(Section*);
 
-class SBLASTER final {
-private:
-	// Data
-	IO_ReadHandleObject read_handlers[0x10]   = {};
-	IO_WriteHandleObject write_handlers[0x10] = {};
-
-	static constexpr auto BlasterEnvVar = "BLASTER";
+void SBLASTER::SetupEnvironment()
+{
+	// Ensure our port and addresses will fit in our format widths.
+	// The config selection controls their actual values, so this is
+	// a maximum-limit.
+	assert(sb.hw.base < 0xfff);
+	assert(sb.hw.irq <= 12);
+	assert(sb.hw.dma8 < 10);
+
+	char blaster_env_val[] = "AHHH II DD HH TT";
```

Comment:
That's what I say when I'm trying to order my tenth beer 😎 🍻

Reply:
:rofl: